File: DB Project/divms/examples/sample_psql.py
import pgdb

s = """ %s """

def index(req):
   message = "<html><head><title>Sample Python Script</title></head>\n"
   # connect
   dbcnx = pgdb.connect(host="dbdev.cs.uiowa.edu:5432:dbdevtest",user="dbdevtest",password="XiuyD=jP")
   
   sqlquery="""SELECT ename, salary, dno FROM Emp;"""
   
   # create a database cursor
   cursor = dbcnx.cursor()
   
   # execute SQL select 
   cursor.execute(sqlquery)
   
   message += "<h2> %s </h2>"
   message %= sqlquery
   message += "<table><tr><th>Name</th><th>Salary</th><th>dno</th></tr>\n"
   
   # get the number of rows in the resultset
   numrows = int(cursor.rowcount)
   
   # get and display one row at a time
   for x in range(0,numrows):
       row = cursor.fetchone()
       message += "<tr><td> %s </td>\n"
       message %= row[0]
       message += "    <td> %s </td>\n"
       message %= row[1]
       message += "    <td> %s </td></tr>\n"
       message %= row[2]
   
   message += "</table>\n"
   message += "</body></html>\n"
   return s % (message)
   
       

File: DB Project/divms/examples/sample_psql.php
<?php
echo("<html><head><title>Sample PHP Script</title></head>\n");

$dbcnx = pg_connect("host='dbdev.cs.uiowa.edu' port='5432' dbname='dbdevtest' user='dbdevtest' password='XiuyD=jP'");
if (!$dbcnx) {
  echo( "<P>Unable to connect to the " .
        "database server at this time.</P>\n" );
  exit();
}
# do a simple query and print it out
// Prepare a query for execution
$sql_query="SELECT ename, salary, dno FROM Emp";

$result_set = pg_query($dbcnx, $sql_query);
if (!$result_set) {
  echo("<P>Error performing query: " .
       pg_result_error() . "</P>");
  exit();
}


echo("<h2>" . $sql_query . "</h2>");
echo("<table><tr><th>Name</th><th>Salary</th><th>dno</th></tr>\n");

while ( $row = pg_fetch_array($result_set) ) {
  echo("<tr><td>" . $row["ename"] . "</td>\n" .
       "     <td>" . $row["salary"] . "</td>\n" .
       "     <td>" . $row["dno"] . "</td>\n" .
       "</tr>\n");
}
echo ("</table>\n");
echo ("</body></html>");
?>
